### PR TITLE
Add ipfs config edit to cli

### LINKED ISSUE #49: config
#### To learn more about config, you can

- try out the ipfs config command with a go-ipfs daemon
- check the config http-api-spec https://github.com/ipfs/http-api-spec/pull/29

#### Implementation Roadmap

- [x] jsipfs config           - Gets or sets the value of a given key of the config file.
  - [x] core
  - [x] http-api (#72)
  - [x] cli
- [x] jsipfs config show           - Outputs the content of the config file.
  - [x] core
  - [x] http-api (#74)
  - [x] cli
- [x] jsipfs config edit           - Opens the config file for editing in $EDITOR.
  - [x] cli (#77)
- [x] jsipfs config replace <file> - Replaces the config with <file>.
  - [x] core
  - [x] http-api (#75)
  - [x] cli

### PR DESCRIPTION
**Pull request by xicombd**
Closes #49

Not sure how to test this though, ideas are welcome :)

### CODE REVIEW

File: src/cli/commands/config/edit.js [hunk 1]
```diff
@@ -26,11 +30,91 @@ module.exports = Command.extend({
   },
 
   run: (name) => {
-    var node = new IPFS()
-    node.version((err, version) => {
-      if (err) { return log.error(err) }
+    const node = new IPFS()
+    const editor = process.env.EDITOR || 'vi'
```
Comment (diasdavid) [inline src/cli/commands/config/edit.js]: if no editor is defined, just say that no editor is defined (same as go-ipfs)

File: src/cli/commands/config/edit.js [hunk 2]
```diff
@@ -26,11 +30,91 @@ module.exports = Command.extend({
...
+
+    async.waterfall([
+      getConfig,
+      saveTempConfig,
+      openEditor,
+      readTempConfig,
+      saveConfig
+    ], (err) => {
+      if (err) {
+        throw err
```
Comment (diasdavid) [inline src/cli/commands/config/edit.js]: throw to where? this will print a stack trace to the user, better print a proper error log

Comment (xicombd) [inline src/cli/commands/config/edit.js]: `ronin` catches it and prints it properly:
screen shot 2016-03-04 at 18 39 31

Comment (diasdavid) [inline src/cli/commands/config/edit.js]: right ronin! :)

File: src/cli/commands/config/edit.js (hunk 1 above)
Comment (xicombd) [inline src/cli/commands/config/edit.js]: yeah I know, this seemed like a better UX to me though. I'll change it :)

**Pull request merged**